admin_manual/configuration/server/security/oauth2.rst

```diff
@@ -0,0 +1,174 @@
+``state``          no        Can be set by the client "to maintain state between the request and callback".
+                             See `RFC 6749`_ for more information.
+================= ========== ========================================================================================
+
+For further information about client registration, please refer to `the official authorization request RFC from the IETF`_.
+
+Authorization Response
+^^^^^^^^^^^^^^^^^^^^^^^
+
+After the resource owner's authorization, the app redirects to the `redirect_uri` specified in the authorization request and adds the authorization code as URL parameter `code`.
+An authorization code is valid for 10 minutes.
+For further information about client registration, please refer to `the official authorization response RFC from the IETF`_.
+
+Access Token Request
+^^^^^^^^^^^^^^^^^^^^
+
+With the authorization code, the client can request an access token using the access token URL.
+`Client authentication`_ is done using basic authentication with the client identifier as username and the client secret as a password.
+The following URL parameters have to be specified:
+
+================= ============================= ===================================================
+Parameter         Required                      Description
+================= ============================= ===================================================
+``grant_type``                                  Either ``authorization_code`` or ``refresh_token``.
+``code``          if the grant type
+                  `authorization_code` is used.
+``redirect_uri``  if the grant type
+                  `authorization_code` is used.
+``refresh_token`` if the grant type
+                  `refresh_token` is used.
+================= ============================= ===================================================
+
+For further information about client registration, please refer to `the official access token request RFC from the IETF`_.
+
+Access Token Response
+^^^^^^^^^^^^^^^^^^^^^
+
+The app responses to a valid access token request with a JSON response like the following.
+An access token is valid for 1 hour and can be refreshed with a refresh token.
+
+.. code-block:: json
+
+  {
+      "access_token" : "1vtnuo1NkIsbndAjVnhl7y0wJha59JyaAiFIVQDvcBY2uvKmj5EPBEhss0pauzdQ",
+      "token_type" : "Bearer",
+      "expires_in" : 3600,
+      "refresh_token" : "7y0wJuvKmj5E1vjVnhlPBEhha59JyaAiFIVQDvcBY2ss0pauzdQtnuo1NkIsbndA",
+      "user_id" : "admin",
+      "message_url" : "https://www.example.org/owncloud/index.php/apps/oauth2/authorization-successful"
+  }
+
+For further information about client registration, please refer to `the official access token response RFC from the IETF`_.
+
+Installation
+------------
+
+To install the application, place the content of the OAuth2 app inside your installation's ``app`` directory.
```

Comment:
Respectively use the Market App

Reply:
As with other ownCloud apps we do not have to document the installation process as it's always the same. Either you install it manually as described or you use the market app. IMO we only need to explain installation when there are dependencies on other software like Collabora or full text search or stuff like that.